code/defines/procs/announcer_datum.dm

```diff
@@ -83,6 +83,9 @@ GLOBAL_DATUM_INIT(major_announcement, /datum/announcer, new(config_type = /datum
 		for(var/mob/M in GLOB.player_list)
 			if(!isnewplayer(M) && M.client)
 				receivers |= M
+			if(!M.say_understands(null, message_language))
+				receivers -= M
+				garbled_receivers |= M
```

Comment:
instead of garbling the message, shouldn't it show as the language spoken instead?